Src/FluentAssertions/DataRowCollectionAssertionExtensions.cs

```diff
@@ -0,0 +1,347 @@
+        /// See the documentation for <see cref="DataRowAssertions{TDataRow}"/> for information about when <see cref="DataRow"/>
+        /// objects are considered equivalent.
+        /// </remarks>
+        /// <param name="expected">The expected <see cref="DataRow"/>.</param>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <paramref name="because"/>.
+        /// </param>
+        public static AndConstraint<GenericCollectionAssertions<DataRow>> ContainEquivalentOf(this GenericCollectionAssertions<DataRow> assertion, DataRow expected, string because = "",
+            params object[] becauseArgs)
+        {
+            return ContainEquivalentOf(assertion, expected, config => config, because, becauseArgs);
+        }
+
+        /// <summary>
+        /// Asserts that a <see cref="DataRowCollection"/> contains at least one object equivalent to another object.
+        /// </summary>
+        /// <remarks>
+        /// See the documentation for <see cref="DataRowAssertions{TDataRow}"/> for information about when <see cref="DataRow"/>
+        /// objects are considered equivalent.
+        /// </remarks>
+        /// <param name="expected">The expected <see cref="DataRow"/>.</param>
+        /// <param name="config">
+        /// A reference to the <see cref="EquivalencyAssertionOptions{DataRow}"/> configuration object that can be used
+        /// to influence the way the object graphs are compared. You can also provide an alternative instance of the
+        /// <see cref="EquivalencyAssertionOptions{DataRow}"/> class. The global defaults are determined by the
+        /// <see cref="AssertionOptions"/> class.
+        /// </param>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <paramref name="because"/>.
+        /// </param>
+        public static AndConstraint<GenericCollectionAssertions<DataRow>> ContainEquivalentOf(this GenericCollectionAssertions<DataRow> assertion, DataRow expected, Func<EquivalencyAssertionOptions<DataRow>,
+                EquivalencyAssertionOptions<DataRow>> config, string because = "", params object[] becauseArgs)
+        {
+            Guard.ThrowIfArgumentIsNull(config, nameof(config));
+
+            if (assertion.Subject is null)
+            {
+                Execute.Assertion
```

Comment:
This needs to take into account if `ContainEquivalentOf` is wrapped in `AssertionScope`

Reply:
I think I see how to do that -- instead of an `if`, always do the assertion, using `ForCondition` to check that the `Subject is not null`, and capture the result of that in e.g. `bool success`. Then, if `success == false` then I know `Subject` is in fact `null`.

Reply:
Moot point now that `ContainEquivalentOf` doesn't exist here any more. :-)